src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -2218,6 +2223,137 @@ static void refda_adm_ietf_dtnma_agent_ctrl_ensure_sbr(refda_ctrl_exec_ctx_t *ct
      * |START CUSTOM FUNCTION refda_adm_ietf_dtnma_agent_ctrl_ensure_sbr BODY
      * +-------------------------------------------------------------------------+
      */
+    const cace_ari_t *odm_ns           = refda_ctrl_exec_ctx_get_aparam_index(ctx, 0);
+    const cace_ari_t *ari_obj_name     = refda_ctrl_exec_ctx_get_aparam_index(ctx, 1);
+    const cace_ari_t *ari_obj_enum     = refda_ctrl_exec_ctx_get_aparam_index(ctx, 2);
+    const cace_ari_t *ari_action       = refda_ctrl_exec_ctx_get_aparam_index(ctx, 3);
+    const cace_ari_t *ari_condition    = refda_ctrl_exec_ctx_get_aparam_index(ctx, 4);
+    const cace_ari_t *ari_min_interval = refda_ctrl_exec_ctx_get_aparam_index(ctx, 5);
+    const cace_ari_t *ari_max_count    = refda_ctrl_exec_ctx_get_aparam_index(ctx, 6);
+    const cace_ari_t *ari_init_enabled = refda_ctrl_exec_ctx_get_aparam_index(ctx, 7);
+
+    cace_ari_t ari_result;
+    cace_ari_init(&ari_result);
+    cace_ari_set_uint(&ari_result, 0);
+
+    refda_agent_t *agent = ctx->runctx->agent;
+    REFDA_AGENT_LOCK(agent, );
+
+    cace_amm_obj_ns_t *odm = cace_amm_obj_store_find_ns(&(agent->objs), odm_ns);
+
+    if (!odm)
+    {
+        CACE_LOG_INFO("ODM not found");
+        return;
```

Comment:
Returning early like this will both leak the stack vars (`ari_result`) but worse will leave the agent lock. A good pattern followed elsewhere is to have an `bool valid = true`, guard each logical section with `if (valid) { ... }`, and if there are any failures set valid to false. Could use other state similarly, but cannot just return early.

Alternately, these conditions can be checked earlier on before allocating stack and taking agent lock.

Reply:
Updated the logic to avoid keeping the lock or leaking memory. Please let me know if you have concerns with the new version.